File: languages/csharp/CommandRunner.cs
namespace Bitwarden.Sdk;
using System.Text.Json;

internal class CommandRunner
{

    private readonly BitwardenSafeHandle _handle;

    internal CommandRunner(BitwardenSafeHandle handle)
    {
        _handle = handle;
    }

    internal T? RunCommand<T>(Command command)
    {
        var req = command.ToJson();
        var result = BitwardenLibrary.RunCommand(req, _handle);
        return JsonSerializer.Deserialize<T>(result);
    }
}

File: languages/go/command_runner.go
package sdk

import (
	"encoding/json"

	"github.com/bitwarden/sdk/languages/go/internal/cinterface"
)

type CommandRunnerInterface interface {
	RunCommand(command Command) (string, error)
}

type CommandRunner struct {
	client cinterface.ClientPointer
	lib    cinterface.BitwardenLibrary
}

func NewCommandRunner(client cinterface.ClientPointer, lib cinterface.BitwardenLibrary) *CommandRunner {
	return &CommandRunner{
		client: client,
		lib:    lib,
	}
}

func (c *CommandRunner) RunCommand(command Command) (string, error) {
	commandJSON, err := json.Marshal(command)
	if err != nil {
		return "", err
	}

	responseStr, err := c.lib.RunCommand(string(commandJSON), c.client)
	if err != nil {
		return "", err
	}

	return responseStr, nil
}
